api/src/main/java/com/sap/xsa/security/container/XSUserInfo.java

```diff
@@ -27,149 +27,205 @@
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBACCOUNT_ID)}
+	 *             from the {@code com.sap.cloud.security.token} package.
 	 * @return subaccount identifier
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubaccountId() throws XSUserInfoException;
 
 	/**
 	 * Return the subdomain of this subaccount
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.SUBDOMAIN)} from
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return subdomain
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getSubdomain() throws XSUserInfoException;
 
 	/**
 	 * Return the client id of the authentication token
-	 * 
+	 *
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.CLIENT_ID)} from
+	 *             the {@code com.sap.cloud.security.token} package.
 	 * @return client id
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getClientId() throws XSUserInfoException;
 
+	/**
+	 * @deprecated Can be replaced with {@code token.getClaimAsString(attribute)}
+	 *             from the {@code com.sap.cloud.security.token} package.
+	 */
 	@Deprecated
 	public String getJsonValue(String attribute) throws XSUserInfoException;
 
 	/**
 	 * Return the email of the user
 	 * 
+	 * @deprecated Can be replaced with
+	 *             {@code token.getClaimAsString(TokenClaims.XSUAA.EMAIL)} from the
+	 *             {@code com.sap.cloud.security.token} package.
+	 * 
 	 * @return email
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	@Deprecated
 	public String getEmail() throws XSUserInfoException;
 
 	@Deprecated // use getHdbToken
 	public String getDBToken() throws XSUserInfoException;
 
+	// TODO 14.01.20 c5295400: can this be replaced?
 	public String getHdbToken() throws XSUserInfoException;
 
 	/**
 	 * Return authentication token
-	 * 
+	 *
+	 * @deprecated Can be replaced with {@code token.getAccessToken()} from the
+	 *             {@code com.sap.cloud.security.token} package.
 	 * @return authentication token
 	 */
+	@Deprecated
 	public String getAppToken();
 
+	// TODO 14.01.20 c5295400: can this be replaced?
 	@Deprecated
 	public String getToken(String namespace, String name) throws XSUserInfoException;
 
 	/**
 	 * Return user attributes
-	 * 
+	 *
 	 * @param attributeName
 	 *            name of attribute
 	 * @return attribute values array
 	 * @throws XSUserInfoException
 	 *             if attribute is not available in the authentication token
 	 */
+	// TODO 14.01.20 c5295400: can this be replaced?
```

Comment:
Maybe we can explain it accordingly:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/master/spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/XsuaaToken.java#L188

Reply:
Okay, I add TODOS accordingly.